iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,460 @@
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected ConfigurationsClient()
+        {
+        }
+
+        internal ConfigurationsClient(IotHubConnectionProperties connectionProperties, HttpTransportSettings transportSettings)
+        {
+            _httpClientHelper = new HttpClientHelper(
+                connectionProperties.HttpsEndpoint,
+                connectionProperties,
+                ExceptionHandlingHelper.GetDefaultErrorMapping(),
+                s_defaultOperationTimeout,
+                transportSettings.Proxy,
+                transportSettings.ConnectionLeaseTimeoutMilliseconds);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration)
+        {
+            return CreateAsync(configuration, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Creates a new Configuration for Azure IoT Edge in IoT hub
+        /// </summary>
+        /// <param name="configuration">The Configuration object being created.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public virtual Task<Configuration> CreateAsync(Configuration configuration, CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Adding configuration: {configuration?.Id}", nameof(CreateAsync));
+
+            try
+            {
+                EnsureInstanceNotClosed();
+
+                if (!string.IsNullOrEmpty(configuration.ETag))
+                {
+                    throw new ArgumentException(ApiResources.ETagSetWhileCreatingConfiguration);
+                }
+
+                var errorMappingOverrides = new Dictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>>
+                {
+                    {
+                        HttpStatusCode.PreconditionFailed,
+                        async responseMessage => new PreconditionFailedException(
+                            await ExceptionHandlingHelper.GetExceptionMessageAsync(responseMessage).ConfigureAwait(false))
+                    }
+                };
+
+                return _httpClientHelper.PutAsync(GetConfigurationRequestUri(configuration.Id), configuration, PutOperationType.CreateEntity, errorMappingOverrides, cancellationToken);
```

Comment:
Let's fix this old problem while we're here.

```suggestion
                return await _httpClientHelper.PutAsync(GetConfigurationRequestUri(configuration.Id), configuration, PutOperationType.CreateEntity, errorMappingOverrides, cancellationToken);
```